app/src/main/java/org/p2p/wallet/solend/ui/earn/bottomsheet/SolendTopUpBottomSheetPresenter.kt

```diff
@@ -0,0 +1,38 @@
+package org.p2p.wallet.solend.ui.earn.bottomsheet
+
+import kotlinx.coroutines.launch
+import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.solend.model.SolendDepositToken
+import org.p2p.wallet.user.interactor.UserInteractor
+
+class SolendTopUpBottomSheetPresenter(
+    private val deposit: SolendDepositToken,
+    private val userInteractor: UserInteractor
+) : BasePresenter<SolendTopUpBottomSheetContract.View>(),
+    SolendTopUpBottomSheetContract.Presenter {
+
+    private var currentToken: Token? = null
+
+    override fun attach(view: SolendTopUpBottomSheetContract.View) {
+        super.attach(view)
+        launch {
+            currentToken = userInteractor.getTokensForBuy(listOf(deposit.tokenSymbol)).firstOrNull()
+            if (currentToken == null) {
+                view.showErrorMessage(IllegalStateException("Token == null!"))
```

Comment:
maybe some other error handling?

Reply:
actually there should not be this case but in code it can =)